src/components/LayoutPage.tsx

```diff
@@ -69,78 +123,319 @@ export const LayoutPage: React.FC = () => {
+                  className="flex items-center justify-between p-3 bg-brown-900 rounded border border-brown-700 hover:border-brown-600"
+                >
+                  <div>
+                    <div className="font-medium text-brown-100">
+                      {machineType.name}
+                    </div>
+                    <div className="text-xs text-brown-400">
+                      {machineType.description}
+                    </div>
+                  </div>
+                  <div className="flex items-center gap-2">
+                    {machineIds.length > 1 && (
+                      <span className="text-sm text-brown-300">
+                        ×{machineIds.length}
+                      </span>
+                    )}
+                    <button
+                      className={`button text-sm ${
+                        isSelected ? "bg-brown-600" : ""
+                      }`}
+                      onClick={() =>
+                        setPlacementMode({
+                          machineType,
+                          machineTypeId: machineIds[0],
+                          rotation: 0,
+                        })
+                      }
+                      disabled={editMode !== "none"}
+                    >
+                      {isSelected ? "Placing..." : "Place"}
+                    </button>
+                  </div>
+                </li>
+              );
+            })}
+          </ul>
+        )}
+      </div>
     </section>
   );
 };
+
+const GhostMachinePreview: React.FC<{
+  placementMode: PlacementMode;
+  cellMap: CellMap;
+  hoverPosition: [number, number] | null;
+}> = ({ placementMode, cellMap, hoverPosition }) => {
+  const isValid =
+    hoverPosition && cellMap.has(hoverPosition)
+      ? canPlaceMachine(
+          cellMap,
+          placementMode.machineType,
+          hoverPosition,
+          placementMode.rotation,
+        )
+      : false;
+
+  // Create a temporary machine state for rendering
+  const ghostMachine = hoverPosition
+    ? {
+        machineTypeId: placementMode.machineTypeId,
+        position: hoverPosition,
+        rotation: placementMode.rotation,
+        selectedOperationId:
+          placementMode.machineType.operations.length > 0
+            ? placementMode.machineType.operations[0].id
+            : "none",
+        operationProgress: {
+          status: "notStarted" as const,
+          ticksRemaining: 0,
+        },
+        inputMaterials: [],
+        processingMaterials: [],
+        outputMaterials: [],
+      }
+    : null;
+
+  // Render ghost preview - non-interactive, just visual
+  if (!ghostMachine || !hoverPosition || !cellMap.has(hoverPosition)) {
+    return null;
+  }
+
+  return (
+    <pixiContainer
+      alpha={0.6}
+      tint={isValid ? 0xffffff : 0xff6666}
+      eventMode="none"
+    >
+      <MachineSprite machine={new Machine(ghostMachine)} />
+    </pixiContainer>
+  );
+};
+
+const GhostMovePreview: React.FC<{
```

Comment:
How are this (`GhostMovePreview`) and `GhostMachinePreview` different? They seem like they should basically be the same thing, a transparent rendering of a machine at a location. I think we should only have one component for rendering a machine ghost.